frameworks/AutoPyTorch/exec.py

```diff
@@ -0,0 +1,178 @@
+import logging
+import os
+import warnings
+import sys
+import numpy as np
+warnings.simplefilter("ignore")
+
+if sys.platform == 'darwin':
+    os.environ['OMP_NUM_THREADS'] = '1'
+
+import pandas as pd
+
+from autoPyTorch import __version__
+from autoPyTorch.api.time_series_forecasting import TimeSeriesForecastingTask
+from autoPyTorch.datasets.time_series_dataset import TimeSeriesSequence
+
+from frameworks.shared.callee import call_run, result
+from frameworks.shared.utils import Timer
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    log.info(f"\n**** AutoPyTorchTS [v{__version__}] ****\n")
+
+    id_column = dataset.id_column
+    prediction_length = dataset.forecast_horizon_in_steps
+    target_column = dataset.target.name
+    seasonality = dataset.seasonality
+
+    dataset_train = pd.concat([dataset.train.X, dataset.train.y], axis=1)
+    dataset_test = pd.concat([dataset.test.X, dataset.test.y], axis=1)
+
+    forecast_horizon = dataset.forecast_horizon_in_steps
+    eval_metric = get_eval_metric(config)
+
+    y_test_future = [seq[1][dataset.target.name].reset_index(drop=True)[-forecast_horizon:] for seq in list(dataset_test.groupby(dataset.id_column))]
+    y_train = [seq[1][dataset.target.name].reset_index(drop=True) for seq in list(dataset_train.groupby(dataset.id_column))]
+
+    X_train = None
+    X_test = None
+
+    known_future_features = None
+
+    start_times_train = [seq[1][dataset.timestamp_column].iloc[0] for seq in list(dataset.train.X.groupby(dataset.id_column))]
+
+    log.info(f'Target name {dataset.target.name}. Forecast horizon {forecast_horizon}.')
+    log.info(f'There are {len(list(dataset_test.groupby(dataset.id_column)))} sequences in the test dataset.')
+    log.info(f'There are {len(list(dataset_train.groupby(dataset.id_column)))} sequences in the train dataset.')
+
+    item_ids =  dataset.test.X[dataset.id_column].unique()
+    items_indices_timestamp = [dataset.test.X[dataset.test.X[dataset.id_column] == item_id].set_index(dataset.timestamp_column).index for item_id in item_ids[:100]]
+    items_freqs = [item_id_indices_timestamp.freq or item_id_indices_timestamp.inferred_freq for item_id_indices_timestamp in items_indices_timestamp]
+    items_freqs_unique = set(items_freqs)
+    if not len(items_freqs_unique) == 1:
+        msg=f"Found not exactly one frequency across all items. Unique inferred frequencies are {items_freqs_unique}"
+        raise ValueError(msg)
+    freq = items_freqs[0]
+
+    with Timer() as training:
+        # initialise Auto-PyTorch api
+        api = TimeSeriesForecastingTask()
+
+        # Search for an ensemble of machine learning algorithms
+        api.search(
+            X_train=X_train,
+            y_train=y_train,
+            X_test=X_test,
+            optimize_metric=eval_metric,
+            n_prediction_steps=forecast_horizon,
+            memory_limit=30 * 1024,  # Currently, forecasting models use much more memories
```

Comment:
I have a note on this in `GluonTS/exec.py`, please explicitly log warnings where you deviate constraints from those defined in the configuration (time, memory, cpu, metric).